src/sema/tests/mod.rs

```diff
@@ -128,3 +139,220 @@ fn test_statement_reachable() {
+            // value 133 does not fit into type int8.
+            int8 add_ovf = 127 + 6;
+    
+            // negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.
+            uint8 negative = 3 - 4;
+    
+            // value 133 does not fit into type int8.
+            int8 mixed = 126 + 7 + input;
+    
+            // negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.
+            return 1 - 2;
+        }
+    
+        function test_mul(int8 input) public {
+            // value 726 does not fit into type int8.
+            int8 mul_ovf = 127 * 6;
+    
+            // value 882 does not fit into type int8.
+            int8 mixed = 126 * 7 * input;
+        }
+    
+        function test_shift(int8 input) public {
+            // value 128 does not fit into type int8.
+            int8 mul_ovf = 1 << 7;
+    
+            // value 128 does not fit into type int8.
+            int8 mixed = (1 << 7) + input;
+        }
+    
+        function test_call() public {
+            // negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.
+            // value 129 does not fit into type int8.
+            test_params(1 - 2, 127 + 2);
+    
+            // negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.
+            // value 129 does not fit into type int8.
+            test_params({usesa: 1 - 2, sesa: 127 + 2});
+        }
+
+        function test_builtin (bytes input) public{
+
+            // value 4294967296 does not fit into type uint32.
+            int16 sesa = input.readInt16LE(4294967296);
+        }
+
+        function test_for_loop () public {
+            for (int8 i = 125 + 5; i < 300 ; i++) {
+            }
+        }
+
+        function composite(int8 a, bytes input) public{
+
+            uint8 sesa = 500- 400 + test_params(100+200, 0) + (200+101) + input.readUint8(4294967296);
+            int8 seas = (120 + 120) + a + (120 + 125);  
+
+            // no diagnostic
+            uint8 b = 255 - 255/5 ;
+
+            // value 260 does not fit into type uint8.
+            uint8 shift_r = (120 >> 2) + 230;
+
+            // value 261 does not fit into type uint8.
+            uint8 mod_test = 254 + (500%17);
+
+            // value 269 does not fit into type uint8.
+            uint8 bb = 320 - (255/5) ;
+        }
+    }
+    
+        "#;
+    let ns = parse(file);
+    let errors = ns.diagnostics.errors();
+
+    assert_eq!(errors[0].message, "value 133 does not fit into type int8.");
+    assert_eq!(errors[1].message, "negative value -1 does not fit into type uint8. Cannot implicitly convert signed literal to unsigned type.");
+    assert_eq!(errors[2].message, "value 133 does not fit into type int8.");
+    assert_eq!(errors[4].message, "value 762 does not fit into type int8.");
```

Comment:
What does `errors[3].message` contain?

Reply:
The same as errors[1], I left it out by mistake